File: 009linkedListValues.js
const a = new Node("a");
const b = new Node("b");
const c = new Node("c");
const d = new Node("d");

a.next = b;
b.next = c;
c.next = d;

// a -> b -> c -> d


const linkedListValues = (head) => {
  // todo
  const result = [];
  let current = head;
  while (current !== null) {
    result.push(current.val);
    current = current.next;
  }
  return result;
};

linkedListValues(a); // -> [ 'a', 'b', 'c', 'd' ]

File: 009linked_list_values.py
# class Node:
#   def __init__(self, val):
#     self.val = val
#     self.next = None

# iterative solution
def linked_list_values(head):
  values = []
  current = head

  while current is not None:
    values.append(current.val)
    current = current.next

  return values

# recursive solution
def linked_list_values(head):
  values = []
  _linked_list_values(head, values)
  return values

def _linked_list_values(head, values):
  if head is None:
    return
  values.append(head.val)
  _linked_list_values(head.next, values)
